diff --git a/bin.cjs b/bin.cjs
index 07274ea6f53fd54a129b3d338892feda6e995cff..b9510c129361e1b56e67fd4e5f2e67d8aa50c76e 100755
--- a/bin.cjs
+++ b/bin.cjs
@@ -82775,6 +82775,7 @@ var init_push = __esm({
         }
       } catch (e4) {
         console.error(e4);
+        process.exit(1);
       }
     };
     sqlitePush = async (schemaPath, verbose, strict, credentials2, tablesFilter, force, casing2) => {
